File: src/ipc/messages/types/structs/AboutApp.ts
import * as ldm from '../../leapdatamodel';
import * as valuetype from '../../valuetype';
import type * as types from '..';
import { StructBase as _StructBase } from '../../StructBase';
import { LeapError as _LeapError } from '../../LeapError';
import { ValueBase as _ValueBase } from '../../ValueBase';

export class AboutApp extends _StructBase {
    static _type: string = 'about-app';
    static _leapStruct: valuetype.LeapStruct = new valuetype.LeapStruct("about-app", [], [new valuetype.Property("version", new valuetype.ValueType("str", []))]);
    version: string;

    constructor(version: string, ) {
        super();
        this.version = version;
    }

    _verifyAsTypeOrRaise(path: string[], valueType: valuetype.ValueType): void {
        if (valueType.name == AboutApp._type) {
            const structType = AboutApp._leapStruct.applyArgs(valueType.args);
            _ValueBase._verifyValueOrRaise(path.concat(['version']), this.version, structType.props[0].propType);
        } else {
            throw _ValueBase._pathError(path, `expecting ${AboutApp._type}`);
        }
    }

    _toLdm(valueType: valuetype.ValueType): ldm.Value {
        let value = new Map<string, ldm.Value>();
        const structType = AboutApp._leapStruct.applyArgs(valueType.args);
        value.set("version", _ValueBase._valueToLdm(this.version, structType.props[0].propType));
        return new ldm.StructValue(value, valueType);
    }
}

File: src-tauri/src/ipc/messages/types/structs/about_app.rs
use std::collections::HashMap as _HashMap;
#[allow(unused_imports)]
use super::super::super::types;
use super::super::super::value_base::ValueBase as _ValueBase;
use super::super::super::struct_base::StructBase as _StructBase;
use super::super::super::leap_data_model as ldm;
use super::super::super::value_type::ValueType as _ValueType;

#[derive(Clone, Debug)]
pub struct AboutApp {
    pub version: String,
}

impl AboutApp {
    pub fn new(version: String, ) -> Self {
        Self {
            version,
        }
    }
}

impl _ValueBase for AboutApp {
    const _TYPE: &'static str = "about-app";

    fn _applied_args() -> Vec<_ValueType> {
        vec![
        ]
    }

    fn _to_ldm(&self) -> ldm::Value {
        ldm::Value::Struct(self._to_ldm_struct())
    }
}

impl _StructBase for AboutApp {
    fn _to_ldm_struct(&self) -> ldm::StructValue {
        let mut value = _HashMap::new();
        value.insert("version".to_owned(), self.version._to_ldm());
        ldm::StructValue{
            value,
            value_type: _ValueType::new(Self::_TYPE.to_owned(), Self::_applied_args()),
        }
    }
}